lib/data/source/network/spoonacular_api.dart

```
@@ -0,0 +1,41 @@
+import 'package:dio/dio.dart';
+import 'package:flap_app/data/dto/recipe_dto.dart';
+import 'package:flap_app/domain/entity/recipe.dart';
+import 'package:flap_app/domain/entity/request_response.dart';
+
+abstract class SpoonacularApi {
+  final Dio dio;
+  SpoonacularApi({required this.dio});
+
+  Future<RequestResponse<List<Recipe>, DioException>> getRandomRecipes();
+}
+
+class SpoonacularApiImpl implements SpoonacularApi {
+  final Dio dio;
+
+  SpoonacularApiImpl({required this.dio});
+
+  @override
+  Future<RequestResponse<List<Recipe>, DioException>> getRandomRecipes() async {
+    try {
+      Response response;
+      response = await dio.get("");
+
+      final recipeListResponse = (response.data["recipes"] as List)
+          .map((e) => RecipeDto.fromJson(e))
+          .toList();
+      return Future.value(Success(recipeListResponse));
+    } on DioException catch (error) {
+      // The request was made and the server responded with a status code
+      // that falls out of the range of 2xx and is also not 304.
+      if (error.response != null) {
+        //Uses DioException types to print error message
+        print(error);
+        return Future.value(Error(error));
+      } else {
+        // Something happened in setting up or sending the request that triggered an Error
+        return Future.value(Error(error));
```

Since we are calling the same line of code regardless of the result of `error.response != null`, do we even need the if else? Or did you intentionally keep it separate so that we can throw a hardcoded error in the future for the case that the error.response is null?

---

I was conflicted about this as well...error.response is going to be printed out if there's a message from the server about the error ...if there is no message from the server about the error, we'll still throw an Error but it it just won't have the detail printed out. Not sure if its worth this extra if else but I think that is the intent.